### PR TITLE
EVM proof of concept

### PR DESCRIPTION
**Pull request by ramtinms**
## Description

This PR adds EVM proof of concept to the emulator (behind a flag). It updates the version of flow-go to the latest version of flow-go on the master that has POC EVM functionality.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: emulator/blockchain.go [hunk 1]
```diff
@@ -78,7 +77,6 @@ func New(opts ...Option) (*Blockchain, error) {
 
 	b := &Blockchain{
 		storage:                conf.GetStore(),
-		broadcaster:            engine.NewBroadcaster(),
```
Comment (bluesign) [inline emulator/blockchain.go:81]: is this removal intentional ?

File: cmd/emulator/start/start.go [hunk 2]
```diff
@@ -73,6 +73,7 @@ type Config struct {
 	RedisURL                 string        `default:"" flag:"redis-url" info:"redis-server URL for persisting redis storage backend ( redis://[[username:]password@]host[:port][/database] ) "`
 	SqliteURL                string        `default:"" flag:"sqlite-url" info:"sqlite db URL for persisting sqlite storage backend "`
 	CoverageReportingEnabled bool          `default:"false" flag:"coverage-reporting" info:"enable Cadence code coverage reporting"`
+	EVMEnabled               bool          `default:"false" flag:"evm-enabled" info:"enables flow evm"`
```
Comment (turbolent) [inline cmd/emulator/start/start.go:76]:
```suggestion
	EVMEnabled               bool          `default:"false" flag:"evm-enabled" info:"enable EVM support"`
```

File: emulator/script_test.go [hunk 3]
```diff
@@ -110,10 +111,10 @@ func TestExecuteScript_WithArguments(t *testing.T) {
 		require.NoError(t, err)
 
 		scriptWithArgs := `
-			pub fun main(n: Int): Int {
```
Comment (turbolent) [inline emulator/script_test.go:113]: Maybe revert all the mixed whitespace changes in the PR

**Review by turbolent — APPROVED** [review]
Nice! 🚀

File: go.mod [hunk 4]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/logrusorgru/aurora v2.0.3+incompatible
 	github.com/onflow/cadence v0.42.5
 	github.com/onflow/flow-archive v1.3.4-0.20230503192214-9e81e82d4dcc
-	github.com/onflow/flow-go v0.32.4-0.20231111003039-3b479f6b0995
+	github.com/onflow/flow-go v0.32.4-0.20231115172515-c1ec969fd6f2
```
Comment (turbolent) [inline go.mod:18]: Is this updating to a commit on `master` or is this pointing to a branch?

File: emulator/blockchain.go [hunk 5]
```diff
@@ -586,6 +598,8 @@ func configureFVM(blockchain *Blockchain, conf config, blocks *blocks) (*fvm.Vir
 		fvm.WithTransactionFeesEnabled(conf.TransactionFeesEnabled),
 		fvm.WithReusableCadenceRuntimePool(customRuntimePool),
 		fvm.WithEntropyProvider(&dummyEntropyProvider{}),
+		// todo use a configuration flag to enable/disable this
```
Comment (turbolent) [inline emulator/blockchain.go:601]: Isn't this TODO already addressed? The argument that's passed is already the configuration option

File: emulator/blockchain.go [hunk 6]
```diff
@@ -711,6 +725,7 @@ func bootstrapLedger(
 	ctx = fvm.NewContextFromParent(
 		ctx,
 		fvm.WithAccountStorageLimit(false),
+		fvm.WithEVMEnabled(false),
```
Comment (turbolent) [inline emulator/blockchain.go:728]: Should this use the configuration option?

File: emulator/blockchain.go [hunk 7]
```diff
@@ -735,6 +750,7 @@ func configureBootstrapProcedure(conf config, flowAccountKey flowgo.AccountPubli
 	options = append(options,
 		fvm.WithInitialTokenSupply(supply),
 		fvm.WithRestrictedAccountCreationEnabled(false),
+		fvm.WithSetupEVMEnabled(true),
```
Comment (turbolent) [inline emulator/blockchain.go:753]: Should this use the configuration option?

File: go.mod (hunk 4 above)
Comment (ramtinms) [reply go.mod:18]: its updating against a latest version of master

File: emulator/blockchain.go (hunk 6 above)
Comment (ramtinms) [reply emulator/blockchain.go:728]: For this one it should always be false to over come the chicken egg problem. I'll add some docs to it

File: cmd/emulator/start/start.go (hunk 2 above)
Comment (ramtinms) [reply cmd/emulator/start/start.go]: updated.

File: emulator/blockchain.go (hunk 6 above)
Comment (ramtinms) [reply emulator/blockchain.go:730]: added some details.

File: emulator/script_test.go (hunk 3 above)
Comment (ramtinms) [reply emulator/script_test.go]: fixed.

File: emulator/blockchain.go (hunk 5 above)
Comment (ramtinms) [reply emulator/blockchain.go]: yes, removed it.

File: emulator/blockchain.go (hunk 7 above)
Comment (ramtinms) [reply emulator/blockchain.go]: Yes, thanks for catching this.

**Comment by turbolent** [comment]
@ramtinms Looks great! Please follow and fill out the checklist in the description

**Comment by ramtinms** [comment]
@sideninja any final look before I merge?

**Review by sideninja — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

**Pull request merged**